src/sagemaker_containers/collections.py

```diff
@@ -0,0 +1,14 @@
+def split_by_criteria(dictionary, keys):  # type: (dict, set) -> (dict, dict)
```

Comment:
is this function used anywhere except Environment? if not, it should be private method in environment

Reply:
Environment by design is a namedtuple, POCO, POJO, (put your plain class of choice here).
Another design decision is that this library is composed by small functions that can be utilized as well. Private methods and classes are only used when necessary.

I have changes outside the scope of this PR that already use split_by_criteria.

Reply:
discussed offline ok.